### ISSUE TITLE
Question about super-resolution losses

### ISSUE THREAD

**Issue by faroit** [j-rixen/SFSRNet-Audio-samples#2]
Hi @j-rixen, thanks for this great paper. I have a question regarding the loss function you describe in the paper when using to sr any backbone model:

![image](https://user-images.githubusercontent.com/72940/216303465-da4f3e07-71f6-4603-928e-49a721ee22e8.png)

I do understand what you mean by the second and third loss. But i fail to understand the first one.
Here is an example:

```
    def forward(self, mix: torch.Tensor) -> dict[str, torch.Tensor]:
        downsampled_input = self.downsample(mix)
        X = self.encoder(downsampled_input)
        Q = self.separator(X)
        S = self.masker(Q) * X
        y_downsampled = self.decoder(S)
        y = self.upsample(y_downsampled)
        y_sr = self.sr_net(mix, y)

```

- loss 2 is "optimizing the final estimations of the separation": `criterion(y, target)`
- loss 3 is "optimizing the estimations that are calculated before the SR network: `criterion(y, downsampled_target)`
- loss 1 is "optimizing the SR network": I don't know what you mean by this. Would this be a separate training to upsample the targets? e.g. `criterion(sr_net(input_signal, downsampled_target), target)`?

**Comment by faroit** [comment]
also slightly related to https://github.com/jeonchangbin49/MedleyVox/issues/1

**Comment by j-rixen** [comment]
The SR model needs to be trained to perform super-resolution, otherwise it is just part of the separation model and won't impact accuracy much (or at all). 
This "loss 1" refers to the loss that is used to train the SR model for SR. Here's what one training step looks like:
- use the full resolution mixture as well as low resolution "quasi" estimations (meaning mixing together sources at a very high SDR, I think it was ~20-30 dB to simulate the separation outputs) as inputs for the SR model to calculate the upsampled version of the quasi estimations
- calculate loss 1 (SI-SDR) using the upsampled quasi estimations and the original full resolution sources
- using the mixture as the input for the separation model, calculate the estimations of the sources (loss 3, basically just the SepFormer model with some slight changes)
- feed the full resolution mixture as well the estimations produced by the separation system through the SR model (loss 2) for the final estimations

Basically, loss 2 is about the final estimations (which obviously need to be included in the loss calculation). Loss 1 is necessary for the SR model to perform SR (otherwise it would just be part of the separation model). Loss 3 is necessary since the SR model expects waveforms of the estimations as part of its input.

**Comment by faroit** [comment]
Thanks for the clarifications!
